packages/components/modules/payments/web/CheckoutComponent/types.ts

```diff
@@ -0,0 +1,30 @@
+import { IProduct, IStripePaymentMethod } from '../types'
+
+export interface CheckoutComponentProps {
+  paymentMethods: IStripePaymentMethod[]
+  product: IProduct
+  setupClientSecret: string
+  onPaymentSuccess?: (subscriptionId: string) => void
+  onPaymentError?: (error: string) => void
+}
+
+export interface BillingDetails {
+  name?: string
+  address?: {
+    line1?: string
```

Comment:
We are reusing the address type in a lot of places, let's create an interface for address and reuse it on every place you have the address type.